foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -0,0 +1,283 @@
+import com.boozallen.aissemble.upgrade.util.pom.LocationAwareMavenReader;
+import org.apache.maven.model.Build;
+import org.technologybrewery.baton.util.FileUtils;
+import com.boozallen.aissemble.upgrade.util.pom.PomHelper;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.model.InputLocation;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import com.boozallen.aissemble.upgrade.util.pom.PomModifications;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+
+import static com.boozallen.aissemble.upgrade.util.pom.PomHelper.getLocationAnnotatedModel;
+import static com.boozallen.aissemble.upgrade.util.pom.PomHelper.writeModifications;
+
+import static org.technologybrewery.baton.util.FileUtils.getRegExCaptureGroups;
+
+/**
+ * This migration is responsible for migrating pom files using Orphedomos to use the Fabric8 docker-maven-plugin.
+ */
+
+public class OrphedomosToFabric8Migration extends AbstractAissembleMigration{
+    private static final Logger logger = LoggerFactory.getLogger(OrphedomosToFabric8Migration.class);
+    private static final String ORPH_GROUP_ID = "org.technologybrewery.orphedomos";
+    private static final String ORPH_ARTIFACT_ID = "orphedomos-maven-plugin";
+    private static final String FABRIC8_GROUP_ID = "io.fabric8";
+    private static final String FABRIC8_ARTIFACT_ID = "docker-maven-plugin";
+    private static final String EXTRACT_PACKAGING_REGEX = "(<packaging>)orphedomos(<\\/packaging>)";
+    private static final String SECOND_REGEX_GROUPING = "$2";
+    private static final String ROOT_ARTIFACT_ID = "build-parent";
+    private static final String ROOT_GROUP_ID = "com.boozallen.aissemble";
+    private static final String TESTS_DOCKER_PARENT_ARTIFACT_ID = "-tests";
+    private static final String TESTS_DOCKER_ARTIFACT_ID = "-tests-docker";
+    private static final String GROUP_ID = "groupId";
+    private static final String ARTIFACT_ID = "artifactId";
+    private static String imageName;
+    private static String imageVersion;
+
+    /**
+     * Function that returns instance of Plugin class with name equal to the pluginName provided
+     * @param model the model to check
+     * @param pluginName the name of the plugin
+     * @return Plugin with the plugin name specified
+     */
+    private Plugin getPlugin(Model model, String pluginName) {
+        Build build = model.getBuild();
+        if (build != null) {
+            List<Plugin> plugins = build.getPlugins();
+            for (Plugin plugin : plugins) {
+                if ((pluginName).equalsIgnoreCase(plugin.getArtifactId())) {
+                    return plugin;
+                }
+            }
+        }
+        return null;
```

Comment:
Would it make sense to throw an exception if the plugin is not found?  Returning a null kicks the can down the road.

Reply:
Agree - a hard failure here via `BatonException` seems reasonable and forces the fix to occur

Reply:
Spoke with @aaron-gary offline and decided to keep the null checks as-is. Adding the try-catch block was useful in catching the case the plugin does not exist but over complicates this PR/migration. For this migration the plugin can be null but the migration should still execute if the packaging type is set to orphedomos.